fe1-web/src/core/components/__tests__/QrCode.test.tsx

```diff
@@ -0,0 +1,43 @@
+import { render, waitFor } from '@testing-library/react-native';
+import React from 'react';
+
+import QRCode from '../QRCode';
+
+const shortText = '8eFmp65HHy7HjJjfTVqfx9t6x5xNqG6MaqRJ2xQaSPvs5vSj5j5RQjU5aS';
+const longText = `${shortText}x`;
+
+// The type is the one of the warn function
+let tmpConsoleWarn: {
+  (...data: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (...data: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+  (message?: any, ...optionalParams: any[]): void;
+};
+describe('QrCode ', () => {
+  // It is useless to test the contrary since we would wait and validate instantly even it would be wrong
+  it('throws a warning, short text with overlay', () => {
+    tmpConsoleWarn = global.console.warn;
+    global.console.warn = jest.fn();
```

Comment:
You might want to check out `spyOn` for this :) The you also don't need the `tmpConsoleWarn`.

https://jestjs.io/docs/jest-object#jestspyonobject-methodname
https://stackoverflow.com/a/56677834/2897827

Reply:
I already tried it but maybe forgot to use global. (It did not work)

Reply:
Okay, the problem that I had is I did not wait for the warning.